e2e/tests/core/02-client/client_test.go

```diff
@@ -153,6 +154,55 @@ func (s *ClientTestSuite) TestClientUpdateProposal_Succeeds() {
 	})
 }
 
+// TestScheduleIBCUpgrade_Succeeds tests that a governance proposal to schedule an IBC software upgrade is successful.
+func (s *ClientTestSuite) TestScheduleIBCUpgrade_Succeeds() {
+	t := s.T()
+	ctx := context.TODO()
+
+	_, _ = s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 10, chainA, chainB))
+
+	authority, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+	s.Require().NoError(err)
+	s.Require().NotNil(authority)
+
+	var (
+		originalTrustingPeriod time.Duration
+		newTrustingPeriod      time.Duration
+	)
+
+	t.Run("send schedule IBC upgrade message", func(t *testing.T) {
+		authority, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+		s.Require().NoError(err)
+		s.Require().NotNil(authority)
+
+		clientState, err := s.QueryClientState(ctx, chainA, ibctesting.FirstClientID)
+		s.Require().NoError(err)
+
+		originalTrustingPeriod = clientState.(*ibctm.ClientState).TrustingPeriod
+		newTrustingPeriod = originalTrustingPeriod + time.Duration(time.Hour*24)
+
+		upgradedClientState := clientState.(*ibctm.ClientState)
+		upgradedClientState.TrustingPeriod = newTrustingPeriod
+
+		latestHeight := clientState.GetLatestHeight().GetRevisionHeight()
+
+		scheduleUpgradeMsg, err := clienttypes.NewMsgIBCSoftwareUpgrade(
+			authority.String(),
+			types.Plan{
+				Name:   "upgrade-client",
+				Height: int64(latestHeight + 100),
+			},
+			upgradedClientState,
+		)
+		s.Require().NoError(err)
+		s.ExecuteGovProposalV1(ctx, scheduleUpgradeMsg, chainA, chainAWallet, 1)
```

Comment:
I wanted to query if the plan has been stored/upgraded client state was stored, but realised after digging around that the UpgradeKeeper doesn't seem to have a query client. what do you guys think? our `msg_server` tests do perform this check, is that sufficient?

Reply:
another option would be to wait for the upgrade plan to pass (`time.Sleep` the requisite amount of time, then check the trusting periods).

Reply:
You can try to upgrade the client on the counterparty. Hermes might do it automatically. Maybe try starting the relayer and querying the client's status afterward

Reply:
> You can try to upgrade the client on the counterparty. Hermes might do it automatically. Maybe try starting the relayer and querying the client's status afterward

Might need to check if interchaintest has built in the support for client updates. Hermes can do it based on the links in my first comment. But I'm unsure if rly also has the same support, and interchaintest would need to expose an API for it in their relayer interface.

Reply:
We added event emission when the consensus state is written allowing the counterparty client to be upgraded, unsure if hermes started listening for that event

Reply:
I regress on my suggestion, happy to finish the full e2e of upgrading the client in a followup pr as this isn't essential for v8 testing

Reply:
after discussion with colin, will just merge this e2e testing the scope of the `ScheduleIBCSoftwareUpgrade` for now.

Reply:
> I think their cmd would need to change for gov v1, maybe it still uses the legacy v1beta1 appraoch

You're right, @damiannolan: hermes is using [gov v1beta proposal](https://github.com/informalsystems/hermes/blob/b7377772a3c9611cff25c5d8cf7ed9a5a26cf992/crates/relayer/src/upgrade_chain.rs#L71-L168). Let's ask them.